From 67686112db842c1baee43c60e08ac42110c8358c Mon Sep 17 00:00:00 2001
Message-Id: <67686112db842c1baee43c60e08ac42110c8358c.1373271639.git.jdenemar@redhat.com>
From: Guannan Ren <gren@redhat.com>
Date: Wed, 6 Mar 2013 15:03:55 -0700
Subject: [PATCH] tests: add one -device video device testcase

https://bugzilla.redhat.com/show_bug.cgi?id=896604

The testcase is for testing non-fixed PCI address for primary
video device and using video args to -deivce qemu option.
(cherry picked from commit ed6fc41b9cee7a878c15e42799a938911eb95c4c)

Conflicts:
	tests/qemuxml2argvdata/qemuxml2argv-video-device-pciaddr-default.args - commit 81af533 not backported
	tests/qemuxml2argvtest.c - no QEMU_CAPS_VNC
---
 .../qemuxml2argv-video-device-pciaddr-default.args |  8 ++++
 .../qemuxml2argv-video-device-pciaddr-default.xml  | 44 ++++++++++++++++++++++
 tests/qemuxml2argvtest.c                           |  5 +++
 3 files changed, 57 insertions(+)
 create mode 100644 tests/qemuxml2argvdata/qemuxml2argv-video-device-pciaddr-default.args
 create mode 100644 tests/qemuxml2argvdata/qemuxml2argv-video-device-pciaddr-default.xml

diff --git a/tests/qemuxml2argvdata/qemuxml2argv-video-device-pciaddr-default.args b/tests/qemuxml2argvdata/qemuxml2argv-video-device-pciaddr-default.args
new file mode 100644
index 0000000..fd9ebac
--- /dev/null
+++ b/tests/qemuxml2argvdata/qemuxml2argv-video-device-pciaddr-default.args
@@ -0,0 +1,8 @@
+LC_ALL=C PATH=/bin HOME=/home/test USER=test LOGNAME=test QEMU_AUDIO_DRV=none \
+/usr/bin/qemu -S -M pc-0.15 -m 1024 -smp 1 -nodefaults \
+-monitor unix:/tmp/test-monitor,server,nowait -no-acpi -boot c \
+-hda /var/lib/libvirt/images/QEMUGuest1 -usb -vnc 127.0.0.1:-5900 \
+-device qxl-vga,id=video0,vram_size=67108864,bus=pci.0,addr=0x3 \
+-device qxl,id=video1,vram_size=67108864,bus=pci.0,addr=0x4 \
+-device qxl,id=video2,vram_size=67108864,bus=pci.0,addr=0x5 \
+-device virtio-balloon-pci,id=balloon0,bus=pci.0,addr=0x2
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-video-device-pciaddr-default.xml b/tests/qemuxml2argvdata/qemuxml2argv-video-device-pciaddr-default.xml
new file mode 100644
index 0000000..120e7f4
--- /dev/null
+++ b/tests/qemuxml2argvdata/qemuxml2argv-video-device-pciaddr-default.xml
@@ -0,0 +1,44 @@
+<domain type='kvm'>
+  <name>QEMUGuest1</name>
+  <uuid>cdbebdfa-1d6d-65c3-be0f-fd74b978a773</uuid>
+  <memory unit='KiB'>1048576</memory>
+  <currentMemory unit='KiB'>1048576</currentMemory>
+  <vcpu placement='static'>1</vcpu>
+  <os>
+    <type arch='x86_64' machine='pc-0.15'>hvm</type>
+    <boot dev='hd'/>
+  </os>
+  <clock offset='utc'/>
+  <on_poweroff>destroy</on_poweroff>
+  <on_reboot>restart</on_reboot>
+  <on_crash>restart</on_crash>
+  <devices>
+    <emulator>/usr/bin/qemu</emulator>
+    <disk type='file' device='disk'>
+      <driver name='qemu' type='qcow2' cache='none'/>
+      <source file='/var/lib/libvirt/images/QEMUGuest1'/>
+      <target dev='hda' bus='ide'/>
+      <address type='drive' controller='0' bus='0' target='0' unit='0'/>
+    </disk>
+    <controller type='usb' index='0'>
+    </controller>
+    <controller type='ide' index='0'>
+    </controller>
+    <input type='mouse' bus='ps2'/>
+    <graphics type='vnc' port='5900' autoport='yes' listen='127.0.0.1'>
+      <listen type='address' address='127.0.0.1'/>
+    </graphics>
+    <video>
+      <model type='qxl' vram='65536' heads='1'/>
+    </video>
+    <video>
+      <model type='qxl' vram='65536' heads='1' primary='yes'/>
+    </video>
+    <video>
+      <model type='qxl' vram='65536' heads='1'/>
+    </video>
+    <memballoon model='virtio'>
+      <address type='pci' domain='0x0000' bus='0x00' slot='0x02' function='0x0'/>
+    </memballoon>
+  </devices>
+</domain>
diff --git a/tests/qemuxml2argvtest.c b/tests/qemuxml2argvtest.c
index e71e948..4ca02db 100644
--- a/tests/qemuxml2argvtest.c
+++ b/tests/qemuxml2argvtest.c
@@ -841,6 +841,11 @@ mymain(void)
             QEMU_CAPS_DRIVE, QEMU_CAPS_DEVICE, QEMU_CAPS_NODEFCONFIG,
             QEMU_CAPS_IDE_CD, QEMU_CAPS_BLOCKIO);
 
+    DO_TEST("video-device-pciaddr-default",
+            QEMU_CAPS_KVM,
+            QEMU_CAPS_DEVICE, QEMU_CAPS_DEVICE_VIDEO_PRIMARY,
+            QEMU_CAPS_DEVICE_QXL, QEMU_CAPS_DEVICE_QXL_VGA);
+
     VIR_FREE(driver.stateDir);
     virCapabilitiesFree(driver.caps);
     VIR_FREE(map);
-- 
1.8.2.1

